core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -216,6 +216,11 @@ public DeleteFiles newDelete() {
     throw new UnsupportedOperationException("Cannot delete from a metadata table");
   }
 
+  @Override
+  public UpdateTableStatistics updateTableStatistics() {
+    throw new UnsupportedOperationException("Cannot update statistics of a metadata table");
```

Comment:
Could you add this as a default implementation rather than adding it in this class? That way, if anyone is using the interface it doesn't break them.

Reply:
Sure, added default method and removed impl stubs from BaseTable and BaseTransaction.
Impl in BaseMetadataTable stays because it includes "a metadata table" in the exc msg (consistently with all other methods in this class).